tests/unit/utils/test_pycrypto.py

```diff
@@ -86,18 +83,24 @@ def test_gen_hash(self):
             )
             self.assertNotEqual(ret, expected["hashed"])
 
-            with self.assertRaises(ValueError, msg=algorithm):
-                ret = salt.utils.pycrypto.gen_hash(
-                    crypt_salt=invalid_salt,
-                    password=passwd,
-                    algorithm=algorithm,
-                    force=force,
-                )
-                self.assertNotEqual(ret, expected["hashed"])
+        with self.assertRaises(ValueError):
```

Comment:
I don't understand this test change...

Reply:
`gen_hash` raises nothing unless it's `blowfish` algorithm and `salt` is longer that `2` chars. I've checked the `crypt` docs, there's nothing about raising `ValueError` too. So I don't know where did you get this. Maybe on windows?

Reply:
Oh yes. Got it, it's from passlib. I'll update the test accordingly.